### PR TITLE
feature/1352 txcontrol

### PR DESCRIPTION
**Pull request by gentee**

### CODE REVIEW

File: packages/daemons/external_network.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		ok       bool
+		connect  *api.Connect
+		duration time.Duration
+		prevTime time.Time
+	)
+	if err = json.Unmarshal([]byte(syspar.SysString(syspar.ExternalBlockchain)), &external); err != nil {
+		return err
+	}
+	for key, netInfo := range external {
+		if len(netInfo.Interval) < 2 {
```
Comment (dchertkov) [inline packages/daemons/external_network.go]: You can use `time.ParseDuration` for parsing

File: packages/smart/funcs.go [hunk 2]
```diff
@@ -2383,3 +2385,50 @@ func PubToHex(in interface{}) (ret string) {
...
+
+// ExternalNetInfo describes external blockchains
+type ExternalNetInfo struct {
+	URL       string `json:"url"`       // Node address
+	Contract  string `json:"contract"`  // The name of the contract
+	Condition string `json:"condition"` // Access rights
+	Interval  string `json:"interval"`  // Interval of sending data
+}
+
+func SendToNetwork(sc *SmartContract, netName string, params interface{}) (err error) {
```
Comment (dchertkov) [inline packages/smart/funcs.go]: Maybe just work with `*types.Map`?

File: packages/smart/funcs.go [hunk 3]
```diff
@@ -2383,3 +2385,50 @@ func PubToHex(in interface{}) (ret string) {
...
+		out, insertQuery string
+		external         map[string]ExternalNetInfo
+		netInfo          ExternalNetInfo
+		ok               bool
+	)
+	out, err = JSONEncode(params)
+	if err != nil {
+		return err
+	}
+	if err = unmarshalJSON([]byte(syspar.SysString(syspar.ExternalBlockchain)), &external,
```
Comment (dchertkov) [inline packages/smart/funcs.go]: Let's move this to the cache. We will only unmarshall when starting and updating the parameter.

**Review by dchertkov — CHANGES_REQUESTED** [review]

File: packages/daemons/external_network.go (hunk 1 above)
Comment (gentee) [reply packages/daemons/external_network.go]: Fixed

File: packages/smart/funcs.go (hunk 2 above)
Comment (gentee) [reply packages/smart/funcs.go]: Fixed

File: packages/smart/funcs.go (hunk 3 above)
Comment (gentee) [reply packages/smart/funcs.go]: SysString gets the value from the cache. 
This function is not called too often. So, I don't think that unmarshalJSON slows down.

**Review by dchertkov — APPROVED** [review]

**Pull request merged**